cmd/zfs/zfs_main.c

```diff
@@ -4571,11 +4594,23 @@ zfs_do_send(int argc, char **argv)
 
 		err = zfs_send_saved(zhp, &flags, STDOUT_FILENO,
 		    resume_token);
+		if (err != 0 && is_dev_error(errno, STDOUT_FILENO)) {
+			(void) fprintf(stderr,
+			    gettext("Error: Stream cannot be written to /dev/"
+			    " files.\n"));
+
+		}
 		zfs_close(zhp);
 		return (err != 0);
 	} else if (resume_token != NULL) {
-		return (zfs_send_resume(g_zfs, &flags, STDOUT_FILENO,
-		    resume_token));
+		err = zfs_send_resume(g_zfs, &flags, STDOUT_FILENO,
+		    resume_token);
+		if (err != 0 && is_dev_error(errno, STDOUT_FILENO)) {
+			(void) fprintf(stderr,
+			    gettext("Error: Stream cannot be written to /dev/"
+			    " files.\n"));
```

Comment:
Shouldn't the error here and elsewhere be, "Error: Stream cannot be written to /dev/null".  There are other /dev/ files which it could be written, and the kernel change which introduced this should only apply to `/dev/null` and `/dev/zero` devices.

Reply:
Yup! As I alluded to in my other comment, I've refactored it to mention both /dev/zero and /dev/null in my local copy, which I haven't pushed yet because I want to figure out a nicer way to handle the failure with `zfs send -R` than `internal error` and `Aborted`.